File: 94. Binary Tree Inorder Traversal.py
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        self.solve(res, root)
        return res

    def solve(self, res: List[int], root: Optional[TreeNode]) -> None:
        if root:
            self.solve(res, root.left)
            res.append(root.val)
            self.solve(res, root.right)  

File: 94. Binary Tree Inorder Traversal.cpp
#include"LeetCodeLibs.h"

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

    class Solution {
    public:
        vector<int> inorderTraversal(TreeNode* root) {
            vector<int> res;
            solve(res, root);
            return res;
        }

        void solve(vector<int>& res, TreeNode* root){
            if(root){
                solve(res, root->left);
                res.push_back(root->val);
                solve(res, root->right);
            }
        }
    };